sdk/driver/driver.go

```diff
@@ -0,0 +1,147 @@
+package driver
+
+import (
+	"encoding/json"
+	"time"
+
+	"gopkg.in/bblfsh/sdk.v1/manifest"
+	"gopkg.in/bblfsh/sdk.v1/protocol"
+	"gopkg.in/bblfsh/sdk.v1/uast"
+	"gopkg.in/bblfsh/sdk.v1/uast/transformer"
+)
+
+var (
+	// ManifestLocation location of the manifest file. Should not override
+	// this variable unless you know what are you doing.
+	ManifestLocation = "/opt/driver/etc/" + manifest.Filename
+)
+
+// Driver implements a bblfsh driver, a driver is on charge of transforming a
+// source code into an AST and a UAST. To transform the AST into a UAST, a
+// `uast.ObjectToNode`` and a series of `tranformer.Transformer` are used.
+//
+// The `Parse` and `NativeParse` requests block the driver until the request is
+// done, since the communication with the native driver is a single-channel
+// synchronous communication over stdin/stdout.
+type Driver struct {
+	NativeDriver
+
+	m *manifest.Manifest
+	o *uast.ObjectToNode
+	t []transformer.Tranformer
+}
+
+// NewDriver returns a new Driver instance based on the given ObjectToNode and
+// list of transformers.
+func NewDriver(o *uast.ObjectToNode, t []transformer.Tranformer) (*Driver, error) {
+	m, err := manifest.Load(ManifestLocation)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Driver{m: m, o: o, t: t}, nil
+}
+
+// Parse process a protocol.ParseRequest, calling to the native driver. It a
+// parser request is done to the internal native driver and the the returned
+// native AST is transform to UAST.
+func (d *Driver) Parse(req *protocol.ParseRequest) *protocol.ParseResponse {
+	r := &protocol.ParseResponse{}
+
+	start := time.Now()
+	defer func() {
+		r.Elapsed = time.Since(start)
+	}()
+
+	var ast interface{}
+	r.Response, ast = d.doParse(req.Language, req.Content, req.Encoding)
+	if r.Status == protocol.Fatal {
+		return r
+	}
+
+	var err error
```

Comment:
Is this needed? Shouldn't be enough with `r.UAST, err :=` below? (same for the `var ast interface{}` above and below).

Reply:
nope, because r.UAST is already defined.